primary/src/helper.rs

```diff
@@ -63,35 +72,128 @@ impl<PublicKey: VerifyingKey> Helper<PublicKey> {
                     self.process_certificates(certificate_ids, requestor, true)
                         .await;
                 }
+                // A request that another primary sends us to check whether we have
+                // available to serve the payloads for the provided certificate_ids.
+                PrimaryMessage::PayloadAvailabilityRequest {
+                    certificate_ids,
+                    requestor,
+                } => {
+                    self.process_payload_availability(certificate_ids, requestor)
+                        .await;
+                }
                 _ => {
                     panic!("Received unexpected message!");
                 }
             }
         }
     }
 
+    /// Checks for each provided digest of the list of digests they node has
+    /// the corresponding certificate within its storage and the payload as
+    /// well. If yes, then the requestor can follow up and request the batches
+    /// payload from the primary's workers via the standard syncing approach.
```

Comment:
Nit: consider something like:
 "Processes a payload availability request by checking we have the certificate & batch data for each certificate digest in `digests`, and reports on each fully available item in the request in a `PayloadAvailabilityResponse`."